shell/platform/darwin/macos/framework/Source/FlutterMutatorView.mm

```diff
@@ -397,37 +421,15 @@ - (void)applyFlutterLayer:(const FlutterLayer*)layer {
 
   /// Paths in global logical coordinates that need to be clipped to.
   NSMutableArray* paths = RoundedRectClipsFromMutations(masterClip, mutations);
-
-  // Add / remove path clip views depending on the number of paths.
-  while (_pathClipViews.count > paths.count) {
-    NSView* view = _pathClipViews.lastObject;
-    [view removeFromSuperview];
-    [_pathClipViews removeLastObject];
-  }
-
-  NSView* lastView = self;
-
-  for (size_t i = 0; i < paths.count; ++i) {
-    FlutterPathClipView* pathClipView = nil;
-    if (i < _pathClipViews.count) {
-      pathClipView = _pathClipViews[i];
-    } else {
-      pathClipView = [[FlutterPathClipView alloc] initWithFrame:self.bounds];
-      [_pathClipViews addObject:pathClipView];
-      [lastView addSubview:pathClipView];
-    }
-    pathClipView.frame = self.bounds;
-    [pathClipView maskToPath:(__bridge CGPathRef)[paths objectAtIndex:i]
-                  withOrigin:finalBoundingRect.origin];
-    lastView = pathClipView;
```

Comment:
`lastView` starts as `self` and as we iterate through the clip view stack, is updated to the most deeply nested clip view as we go. In the refactor, we just look up the most deeply nested clip view when we need it instead.